src/win/fs.c

```diff
@@ -1911,17 +1869,18 @@ int uv_fs_open(uv_loop_t* loop, uv_fs_t* req, const char* path, int flags,
 
   if (cb) {
     QUEUE_FS_TP_JOB(loop, req);
-    return 0;
   } else {
     fs__open(req);
-    return req->result;
+    if (req->result < 0)
```

Comment:
The old version does a silent truncating cast (from HANDLE to int), https://github.com/libuv/libuv/pull/1166#discussion_r102083106